Send a weekly overview of mail received past week

Justin-Koningsberger opened the pull request.

lucasluitjes commented on lib/tasks/scheduler.rake:
```diff
@@ -1,8 +1,8 @@
 desc "This task is called by the Heroku scheduler add-on"
 task :email_overview => :environment do
-  if (Time.now.wday == 5 && Time.now.hour > 15)
+  if (!(DateTime.parse(Overview.last_overview?.to_s).today?) && Time.now.wday == 5 && Time.now.hour > 15)
```
﻿The logic regarding last_overview makes no sense to me. I think if you add tests you'll find that it doesn't work and can fix it. If you want we can write the test together, and you can fix the implementation solo?

lucasluitjes commented on app/mailers/application_mailer.rb:
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'last-week@overview.com'
```
﻿The convention is to use example.org when developing/testing. In production we should use a domain that we control, not a made up one that probably belongs to someone. The reason is that there are antispam mechanisms using DNS records. So using overview.com when we don't own it likely means our e-mails will end up in the spam folder.

Probably the best thing is to check if we're running in production, and in that case use luitjes.it. If not, use example.org.

Justin-Koningsberger commented:
I changed the email notification to only run on hacker newsletters, moved the email addresses to the dotenv. I removed some hardcoded values from the application.js, and I changed the javascript to use arrow functions. Then I ran Rubocop (I think for the first time?).

After merging, we need to change the Mailgun username and password, add them to Heroku, and change them in our dotenv.

These are the Trello cards regarding the requested features:
https://trello.com/c/BvdfMVlB/6-emailreader-mail-versturen
https://trello.com/c/pGxyp8TJ/4-e-mailadres-showen-in-e-mailreader
https://trello.com/c/TSZLFOyr/17-email-reader

lucasluitjes commented on app/assets/javascripts/application.js:
```diff
@@ -14,106 +14,110 @@
-  else {
-    return true;
-  } 
+  else {true;}
```
Hier staat `true` en de oude versie was `return true`. Ik weet niet zeker of je in javascript mag returnen door de laatste statement te laten staan, dus dat even laten staan.

lucasluitjes commented on app/assets/javascripts/application.js:
```diff
@@ -14,106 +14,110 @@
+  else {true;}
+});
+
+document.addEventListener('keydown', (event) => {
```
Algemene feedback, hoeft niet voor deze PR: verder zie ik op de nieuwe 69, 81, 89 en nog een aantal plekken dat je heel veel key_down event listeners aanmaakt voor elk stukje functionaliteit. Niet voor deze PR, maar het zou netter zijn om daar gewoon één voor te hebben die alles doet, en de inhoud zit dan in losse functies. Dus dan krijg je één event listener met daar een if statement die bepaalt welke functie er moet worden aangeroepen.

lucasluitjes commented on app/controllers/incoming_emails_controller.rb:
```diff
@@ -12,6 +12,8 @@ def create
       body: params[:message]
     )
+
+    SendNotificationJob.perform_now(email)
```
send_notification_job.perform_now. Als ik me niet vergis betekent dat dat hij de code inline gaat draaien ipv in een job.  Dus dat betekent dat de incoming email ook niet goed gaat als er iets mis is met de email service ofzo. Dus die job moet net als de create_links_job met perform_later die op de regel eronder staat.

lucasluitjes commented on app/models/overview.rb:
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
```
Dan zie ik app/models/overview.rb bestaan, ik denk dat die helemaal niet meer nodig is. Dat was geloof ik de oude functionaliteit.
